src/services/createRouter.ts

```diff
@@ -180,61 +181,79 @@ export function createRouter<const TRoutes extends Routes, const TOptions extend
     history.startListening()
   }
 
+  const resolve: RouterResolve<TRoutes> = (
+    source: Url | RoutesName<TRoutes>,
+    paramsOrOptions?: Record<string, unknown>,
+    maybeOptions?: RouterResolveOptions,
+  ) => {
+    if (!isUrl(source)) {
+      const params = paramsOrOptions ?? {}
+      const options: RouterResolveOptions = maybeOptions ?? {}
+      const match = routes.find((route) => route.name === source)
+
+      if (!match) {
+        throw new RouteNotFoundError(String(source))
+      }
```

Comment:
I don't think we need to throw an error do we? RouterResolve's return type is `ResolvedRoute | undefined`. So I think we can just return `undefined` here.